app/Services/Model/Imp/SummitOrderService.php

```
@@ -976,6 +1115,105 @@ public function undo()
+     * @var array
+     */
+    private $payload;
+
+    /**
+     * @var ISummitTicketTypeRepository
+     */
+    private $ticket_type_repository;
+
+    /**
+     * @var ILockManagerService
+     */
+    private $lock_service;
+
+    /**
+     * AutoAssignPrePaidTicketTask constructor.
+     * @param Summit $summit
+     * @param array $payload
+     * @param ISummitTicketTypeRepository $ticket_type_repository
+     * @param ITransactionService $tx_service
+     * @param ILockManagerService $lock_service
+     */
+    public function __construct
+    (
+        Summit $summit,
+        array $payload,
+        ISummitTicketTypeRepository $ticket_type_repository,
+        ITransactionService $tx_service,
+        ILockManagerService $lock_service
+    )
+    {
+        $this->tx_service = $tx_service;
+        $this->lock_service = $lock_service;
+        $this->summit = $summit;
+        $this->payload = $payload;
+        $this->ticket_type_repository = $ticket_type_repository;
+    }
+
+    public function run(array $formerState): array
+    {
+        $this->formerState = $formerState;
+
+        return $this->tx_service->transaction(function () {
+            $tickets = $this->payload['tickets'];
+            $ticket_dto = $tickets[0];
+
+            $order = $this->summit->getOrderById($this->payload['order_id']);
+
+            if (is_null($order)) {
+                throw new EntityNotFoundException('Order not found.');
+            }
+
+            $promo_code_val = $ticket_dto['promo_code'] ?? null;
+            if(!empty($promo_code_val)) {
+                $type_id = $ticket_dto['type_id'];
+                $this->lock_service->lock('ticket_type.' . $type_id . 'promo_code.' . $promo_code_val .'.sell.lock',
+                    function () use ($order, $promo_code_val, $type_id) {
+                        $promo_code = $this->summit->getPromoCodeByCode($promo_code_val);
+                        if (is_null($promo_code)) return ['order' => $order];
```

you should throw entity not found exception here